test/implementations/functions_testing.py

```diff
@@ -8,436 +7,659 @@
     relative_error_tol,
     zero_matrix,
 )
+from reverse_kronecker_product import reverse_kronecker_product
 
 
 # 1.1 No Auxiliary
 def verify_circuit_no_auxiliary(unitary_matrix, controls_no: int, auxiliaries_no: int):
-    # get mct inverse matrix
-    inverse_matrix = load_matrix("noauxiliary", controls_no)
+    try:
+        # get mct inverse matrix
+        inverse_matrix = load_matrix("noauxiliary", controls_no)
 
-    no_of_qubits = controls_no + 1
+        no_of_qubits = controls_no + 1
 
-    # X_1 * X_2^dagger * np.conj((X_1 * X_2^dagger)[0,0]) - I = 0
-    M = np.matmul(unitary_matrix, inverse_matrix)
-    generated_unitary = M * np.conjugate(M[0, 0]) - identity_matrix(no_of_qubits)
+        # X_1 * X_2^dagger * np.conj((X_1 * X_2^dagger)[0,0]) - I = 0
+        M = np.matmul(unitary_matrix, inverse_matrix)
+        generated_unitary = M * np.conjugate(M[0, 0]) - identity_matrix(no_of_qubits)
 
-    # Expected unitary after calculation is 0.
-    expected_unitary = zero_matrix(no_of_qubits)
+        # Expected unitary after calculation is 0.
+        expected_unitary = zero_matrix(no_of_qubits)
 
-    if generated_unitary.shape != expected_unitary.shape:
-        return False, "Unitary has different shape."
+        if generated_unitary.shape != expected_unitary.shape:
+            return False, "Unitary has different shape."
 
-    if (
-        np.allclose(
-            generated_unitary, expected_unitary, atol=absolute_error_tol, rtol=relative_error_tol
-        )
-        is False
-    ):
-        return False, "Something wrong with the implementation"
+        if (
+            np.allclose(
+                generated_unitary,
+                expected_unitary,
+                atol=absolute_error_tol,
+                rtol=relative_error_tol,
+            )
+            is False
+        ):
+            return False, "Something wrong with the implementation"
 
-    return True, ""
+        return True, ""
+    except Exception:

```

Comment:
this cannot be like this. you should check what kind of exception is raised and react accordingly (actually why do you need try-except in the first place?)

Reply:
moved to issue #16